src/crypto_helpers.rs

```diff
@@ -1,15 +1,54 @@
 use nanos_sdk::bindings::*;
 use nanos_sdk::ecc::{CurvesId, DEREncodedECDSASignature};
 
-// TODO: make a macro that derives the path by itself
-// ex: make_bip32_path!(b"44'/535348'/0'/0/0";
-// =>
-pub const BIP32_PATH: [u32; 5] = [
-        0x8000002cu32,
-        0x80082b34, 
-        0x80000000, 
-        0x00000000, 
-        0x00000000];
+/// ASCII value of the character 0.
+const ZERO: u8 = 48;
+/// ASCII value of the character 9.
+const NINE: u8 = 57;
+
+/// Helper macro that creates an array from the ASCII values of a correctly formatted derivation path.
+/// Format expected: `b"44'/coin_type'/account'/change/address"`.
+///
+/// # Panics
+///
+/// Panics if the parameter does not follow the correct format.
+macro_rules! make_bip32_path {
+    ($bytes:expr) => {{
+        // The three first elements must start with `0x800000`.
```

Comment:
Is this even correct? I didn't find a good explanation as to why those three bytes were to start with `0x800000` but just guessed it from the code that was here before.

Reply:
The MSB of a bip32 path element is set to 1 to request a hardened derivation.
Because of bip44, the first 3 elements of the derivation path are hardened (this is represented by the `'` after each index in the ascii path), so they're worth `0x80000000 | index`.
https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki

Reply:
That was my understanding (i.e `'` means "add `0x800000`). Thank you for the links. So IIUC, we can let the code just as is, right?

Reply:
I'll let a rust enabled colleague answer to that ~